frontend/src/components/car-list-item/car-list-item.tsx

```diff
@@ -1,41 +1,79 @@
 import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 
-import { ComplectationDetailsType } from '@autoline/shared/common/types/types';
+import {
+  ComplectationDetailsType,
+  WishlistInput,
+} from '@autoline/shared/common/types/types';
+import { AppRoute } from '@common/enums/enums';
 import { CarListItemProps } from '@common/types/types';
 import { SliderNavButton } from '@components/car-list-item/slider-nav-button/slider-nav-button';
 import { swiperParams } from '@components/car-list-item/swiper-params';
+import { HeartIcon } from '@components/common/icons/icons';
 import { Spinner } from '@components/common/spinner/spinner';
 import { CompleteSetTableCollapsed } from '@components/complete-set-table/complete-set-table-collapsed';
 import { formatPrice } from '@helpers/helpers';
 import { objectToQueryString } from '@helpers/object-to-query';
+import { useAppSelector } from '@hooks/hooks';
 import { Grid } from '@mui/material';
 import { uuid4 } from '@sentry/utils';
 import {
   useGetComplectationsQuery,
   useGetModelDetailsQuery,
 } from '@store/queries/cars';
+import {
+  useGetWishlistsLikedItemQuery,
+  useCreateWishlistMutation,
+  useDeleteWishlistMutation,
+} from '@store/queries/preferences/wishlist';
 import { clsx } from 'clsx';
 import { Swiper, SwiperSlide } from 'swiper/react';
 
 import styles from './styles.module.scss';
 
 const CarListItem: React.FC<CarListItemProps> = (props) => {
   const { model_id, complectations_id } = props;
+  const authToken = useAppSelector((state) => state.auth.token);
```

Comment:
Could you please select a shared selector for the token? I saw that it's extracted like this in a few places